test/unit/lib/challenge.test.ts

```diff
@@ -26,3 +27,173 @@ describe('Challenge Generation', (): void => {
     expect(Buffer.byteLength(challenge, 'base64')).toBe(8)
   })
 })
+
+// Each test generates a random key pair
+describe('Signature Verification', (): void => {
```

Comment:
Can you please add a test case for the error that is being thrown 'Unable to verify signature'?

Reply:
Could you suggest something here @lewisdaly ? I was thinking about this but I didn't know how to get the function to throw an error here.

I just followed the pattern in the existing function from @akapur99 . There is no error test case for that.